opencensus-migration/src/main/java/OpenTelemetryStartEndHandler.java

```diff
@@ -0,0 +1,37 @@
+import io.opencensus.implcore.trace.RecordEventsSpanImpl;
+import io.opencensus.implcore.trace.RecordEventsSpanImpl.StartEndHandler;
+import io.opencensus.trace.Annotation;
+import io.opencensus.trace.export.SpanData;
+import io.opencensus.trace.export.SpanData.TimedEvent;
+import io.opentelemetry.trace.Span;
+
+public class OpenTelemetryStartEndHandler implements StartEndHandler {
+
+  private static final double NANOS_PER_SECOND = 1e9;
+
+  private final SpanCache spanCache;
+
+  public OpenTelemetryStartEndHandler() {
+    this.spanCache = SpanCache.getInstance();
+  }
+
+  @Override
+  public void onStart(RecordEventsSpanImpl ocSpan) {
+    spanCache.toOtelSpan(ocSpan);
+  }
+
+  @Override
+  public void onEnd(RecordEventsSpanImpl ocSpan) {
+    Span span = spanCache.toOtelSpan(ocSpan);
+    SpanData spanData = ocSpan.toSpanData();
+    spanCache.removeFromCache(ocSpan);
+    for (TimedEvent<Annotation> annotation : spanData.getAnnotations().getEvents()) {
+      span.addEvent(
+          annotation.getEvent().getDescription(),
+          (long)
+              (annotation.getTimestamp().getSeconds() * NANOS_PER_SECOND
+                  + annotation.getTimestamp().getNanos()));
+    }
```

Comment:
It seems that OpenTelemetry only supports parent links when a span is first created, so I can't actually map any OpenCensus links added after the OC span is started here

Reply:
Aren't we supposed to add parent link in `SpanConverter.toOtelSpan` (i.e. when we are creating an OT copy)?

Reply:
Yep that's being added in https://github.com/zoercai/opentelemetry-operations-java/pull/3/files#diff-513832629a310343a2179545ed55a9c21e03d67931cb68991ecb34d442d88788R79-R88 when the span is created. But in OpenCensus more links can be added afterwards (like how events are added), which isn't supported by OpenTelemetry.

Reply:
That's probably fine, though it might be worth calling that out in the README once we have it.